gridData/CCP4.py

```diff
@@ -0,0 +1,211 @@
+# gridDataFormats --- python modules to read and write gridded data
+# Copyright (c) 2009-2014 Oliver Beckstein <orbeckst@gmail.com>
+# Released under the GNU Lesser General Public License, version 3 or later.
+
+"""
+:mod:`CCP4` --- the CCP4 volumetric data format
+===========================================
+
+.. _gOpenMol: http://www.ccp4.ac.uk/html/maplib.html#description
+
+The module provides a simple implementation of a reader for CCP4_
+*ccp4* files. CCP4 files are binary files. The :class:`CCP4` reader tries
+to guess the endianess of the file, but this can fail (with a
+:exc:`TypeError`); you are on your own in this case.
+
+Only the reader is implemented. If you want to write gridded data use a format
+that is more standard, such as OpenDX (see :mod:`OpenDX`).
+
+
+Background
+----------
+
+CCP4 format: http://www.ccp4.ac.uk/html/maplib.html#description
+
+Used to be more carefully documented at http://lsbr.niams.nih.gov/3demc/3demc_maplib.html but currently this is only accessible through the Google cache
+http://webcache.googleusercontent.com/search?q=cache:KRSvXB0S3dsJ:lsbr.niams.nih.gov/3demc/3demc_maplib.html
+
+"""
+# TODO: include basic format in documentation.
+
+from __future__ import with_statement
+
+import warnings
+import struct
+import numpy
+
+from gOpenMol import Record
+
+#TODO: Consider abstracting a binary data class to handle CCP4,
+# gOpenMol, and other binary formats.
+class CCP4(object):
+    """A class to represent a CCP4_ file.
+
+    Only reading is implemented; either supply a filename to the constructor
+      >>> G = CCP4(filename)
+    or load the file with the read method
+      >>> G = CCP4()
+      >>> G.read(filename)
+
+    The data is held in :attr:`CCP4.array` and all header information is in
+    the dict :attr:`CCP4.header`.
+
+    :attr:`CCP4.shape`
+         D-tuplet describing size in each dimension
+    :attr:`CCP4.origin`
+         coordinates of the centre of the grid cell with index 0,0,...,0
+    :attr:`CCP4.delta`
+         DxD array describing the deltas
```

Comment:
Can you add a note that says what is currently not implemented (basically, the TODO comments), along the lines of 
```reST
... Note:: The following features of the CCP4 format are *not* implemented:
    * triclinic boxes
    * ...
```